src/SuperVault.sol

```diff
@@ -533,16 +586,58 @@ contract SuperVault is BaseStrategy, ISuperVault {
             }
 
             mvData.maxSlippages[i] = MAX_SLIPPAGE;
+
+            unchecked {
+                ++i;
+            }
         }
 
         return mvData;
     }
 
+    function _prepareSingleVaultWithdrawData(uint256 amount_) internal view returns (SingleVaultSFData memory svData) {
+        svData.superformId = superformIds[0];
+        (address superform,,) = svData.superformId.getSuperform();
+
+        // Get current balances
+        uint256 spBalance = _getSuperPositionBalance(_SUPER_POSITIONS, svData.superformId);
+        uint256 assetBalance = IBaseForm(superform).previewRedeemFrom(spBalance);
+
+        // Set up basic data
+        svData.liqRequest.token = address(asset);
+        bool isERC5115 = _isERC5115Vault(svData.superformId);
+
+        if (isERC5115) {
+            svData.liqRequest.interimToken = address(asset);
+        }
+
+        // Calculate withdrawal amounts
+        if (amount_ >= assetBalance) {
+            svData.amount = spBalance;
+            svData.outputAmount = _tolerance(isERC5115, assetBalance);
+        } else {
+            svData.outputAmount = _tolerance(isERC5115, amount_);
+            svData.amount = IBaseForm(superform).previewDepositTo(amount_);
+
+            if (svData.amount > spBalance) {
+                svData.amount = spBalance;
+            }
+        }
+
+        svData.maxSlippage = MAX_SLIPPAGE;
+        svData.hasDstSwap = false;
+        svData.retain4626 = false;
+        svData.receiverAddress = address(this);
+        svData.receiverAddressSP = address(this);
+
+        return svData;
+    }
+
     /// @notice Checks if a vault is ERC5115 and validates form implementation IDs
     /// @param superformId_ The superform ID to check
     /// @return isERC5115 True if the vault is ERC5115
     function _isERC5115Vault(uint256 superformId_) internal view returns (bool isERC5115) {
-        ISuperformFactoryMinimal factory = ISuperformFactoryMinimal(_getAddress(keccak256("SUPERFORM_FACTORY")));
+        ISuperformFactoryMinimal factory = ISuperformFactoryMinimal(_SUPERFORM_FACTORY);
```

Comment:
Maybe store that fact that a vault is erc5115 in with the superformId (in storage).
Have to test if this actually would safe gas